inc/mailcollector.class.php

```diff
@@ -172,11 +164,10 @@ function post_getEmpty() {
       $this->fields['is_active']    = 1;
    }
 
-
-   /**
-    * @see CommonDBTM::prepareInputForUpdate()
-   **/
-   function prepareInputForUpdate($input) {
+   public function prepareInput(array $input, $mode = 'add') :array {
+      if ('add' === $mode && !isset($input['name']) || empty($input['name'])) {
```

Comment:
For the functionnal point of view, this error does not make any sense. The name of the collector can be an email address or not, it will not prevent the collector to work.

I propose to remove this error.

Reply:
Same remark for line 184.

Reply:
Goal was to factorize checks, I do not want to change any logic here; this will be done on collector rewrite.